src/ucp/wireup/select.c

```diff
@@ -384,6 +398,64 @@ ucp_wireup_select_reg_mem_types(ucp_context_h context, ucp_md_index_t md_index)
     return reg_mem_types;
 }
 
+static ucp_wireup_remote_device_count_t *ucp_wireup_get_remote_device_count(
```

Comment:
the logic of fixing dev_index should be part of address unpack
i would expect select.c code to not change at all